src/v/kafka/client/transport.h

```diff
@@ -206,7 +212,7 @@ class transport : public net::base_transport {
         wr.write(int16_t(key()));
         wr.write(int16_t(version()));
         wr.write(int32_t(_correlation()));
-        wr.write(std::string_view("test_client"));
+        wr.write(_client_id);
```

Comment:
The type has changed. Does the `response_writer` do the right thing, or should this be `_client_id.value_or("")`.

Reply:
I think that's one of the quirks of the protocol, here's a link to the relevant kafka source: https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/requests/RequestHeader.java#L97

Reply:
FWIW redpanda also has a conditional where it will accept -1 as a value for a client id's size

Reply:
```java
if (headerData.clientId() == null) {
    headerData.setClientId("");
}
```
Looks a lot like
```c++
  _client_id.value_or("");
```
It seems to me that the API has it as nullable, but the wire format does not.

Reply:
I think its the other way around, wire format is fine to have -1 (i.e. parsers won't throw) but any entity ever querying client id via a kafka broker will only observe a non nullable type

Reply:
Oh, I see.